File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int maxipro=0,miniprice=prices[0];
        for(int i=1;i<prices.size();i++)
        {
            maxipro=max(maxipro,prices[i]-miniprice);
            miniprice=min(miniprice,prices[i]);
        }
        return maxipro;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int maxProfit=0,minValue=prices[0];
        for(int i=1;i<prices.length;i++)
        {
            int profit=prices[i]-minValue;
            maxProfit=Math.max(maxProfit,profit);
            minValue=Math.min(minValue,prices[i]);
        }
        return maxProfit;
        
    }
}